driver/src/transport.rs

```diff
@@ -109,6 +123,27 @@ impl Transport for HttpTransport {
     }
 
     fn send(&self, id: RequestId, request: Call) -> Self::Out {
-        self.0.clone().execute_rpc(id, request).boxed().compat()
+        self.0
+            .clone()
+            .execute_rpc(id, Request::Single(request))
```

Comment:
@nlordell can you remind me again why we are implementing our own transport instead of wrapping the web3-rs HTTP transport? Was it to have access to the raw response? If so, what about their error wrapping was insufficient for our purposes?

Reply:
https://github.com/gnosis/dex-services/pull/543 , The main reason was to implement timeouts. Being able to log the raw response was a bonus as the error message was always sufficient to figure out the problem. For example if it failed to parse the response (for example when Ganache returns both a result and error), the error was just `"data did not match any variant of untagged enum Response"` and since the error is `.to_string()`-ed in the `web3::Error`, things like parse error position also get lost.

I think wrapping the web3 transport was presented as an option, but we decided to write our own transport as it was fairly simple and it wasn't clear if implementing timeouts correctly was as easy as it would require dealing with timers and dropping the HTTP transport future didn't actually cancel the HTTP request.